### PR TITLE
Convert plots to GGplot

### PR DESCRIPTION
**Pull request by koenderks**

### CODE REVIEW

File: JASP-Engine/JASP/R/descriptives.R [hunk 1]
```diff
@@ -791,138 +800,163 @@ Descriptives <- function(jaspResults, dataset, options, state=NULL)
...
-  if(rugs)
-    rug(jitVar)
+    return(p)
 
-  lines(density$x[density$x>= min(ax1) & density$x <= max(ax1)], density$y[density$x>= min(ax1) & density$x <= max(ax1)], lwd= lwd)
 }
 
 .barplotJASP <- function(column, variable, dontPlotData= FALSE){
+    
+    library(JASPgraphs)
```
Comment (vandenman) [inline JASP-Engine/JASP/R/descriptives.R]: remove and use `JASPgraphs::`?

File: JASP-Engine/JASP/R/descriptives.R [hunk 2]
```diff
@@ -791,138 +800,163 @@ Descriptives <- function(jaspResults, dataset, options, state=NULL)
...
 
-  fit[[1]] <- lm(yy ~ poly(xx, 1, raw= TRUE), d)
-  fit[[2]] <- lm(yy ~ poly(xx, 2, raw= TRUE), d)
-  fit[[3]] <- lm(yy ~ poly(xx, 3, raw= TRUE), d)
-  fit[[4]] <- lm(yy ~ poly(xx, 4, raw= TRUE), d)
+.plotScatterDescriptives <- function(xVar, yVar, cexPoints= 1.3, cexXAxis= 1.3, cexYAxis= 1.3, lwd= 2){
 
-  # find parsimonious, best fitting regression model
-  Bic <- vector("numeric", 4)
+    library(JASPgraphs)
```
Comment (vandenman) [inline JASP-Engine/JASP/R/descriptives.R]: remove and use JASPgraphs::?

File: JASP-Engine/JASP/R/regressionlinear.R [hunk 3]
```diff
@@ -3325,170 +3311,163 @@ RegressionLinear <- function(dataset=NULL, options, perform="run", callback=func
...
-
-	plot(1, xlim= range(xticks), ylim= c(ylow, yhigh), type="n", axes=FALSE, ylab="", xlab="")
-	h <- hist(res, freq=F, main = "", ylim= c(ylow, yhigh), xlab = "", ylab = " ", axes = F, col = "grey", add= TRUE, nbreaks= round(length(res)/5))
-	ax1 <- axis(1, line = 0.3, at= xticks, lab= xticks, cex.axis = 1.2)
-	mtext(text = resName, side = 1, cex=1.5, line = 3)
-	ax2 <- axis(2, at = c(0, max(max(h$density), max(density$y))/2, max(max(h$density), max(density$y))) , labels = c("", "Density", ""), lwd.ticks=0, pos= range(ax1)- 0.05*diff(range(ax1)), cex.axis= 1.5, mgp= c(3, 0.7, 0), las=0)
-
-	if(rugs)
-		rug(jitVar)
+    library(JASPgraphs)
```
Comment (vandenman) [inline JASP-Engine/JASP/R/regressionlinear.R]: remove and use JASPgraphs::?

File: JASP-Engine/JASP/R/common.R [hunk 4]
```diff
@@ -2635,7 +2637,8 @@ editImage <- function(plotName, type, height, width) {
 
 			# copy plot and check if we edit it
 			plot <- oldPlot
-			if (FALSE && type == "interactive" && isGgplot) {
+			if (type == "interactive" && isGgplot) {
+			#if (FALSE && type == "interactive" && isGgplot) {
```
Comment (vandenman) [inline JASP-Engine/JASP/R/common.R]: Could you change it back to `if (FALSE && type == "interactive" && ..`? We wouldn't want any users to accidentally try out this feature :)

**Comment by koenderks** [comment]
Implemented Don's changes

**Comment by koenderks** [comment]
This pull request will be merged into version 0.9.2. Until then it will remain here, not updated. The up-to-date branch for this project is called PLOT_backup on my github page (I will keep it up to date). When the time comes, I will open another pull request from this branch.

**Pull request closed without merging**